packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -0,0 +1,48 @@
+import * as cdk from '@aws-cdk/core';
+import { CfnResource } from '@aws-cdk/core';
+import { TagUpdateConstraintOptions } from '../constraints';
+import { IPortfolio } from '../portfolio';
+import { IProduct } from '../product';
+import { CfnPortfolioProductAssociation, CfnResourceUpdateConstraint } from '../servicecatalog.generated';
+import { hashValues } from './util';
+import { InputValidator } from './validation';
+
+export class AssociationManager {
+  public static associateProductWithPortfolio(portfolio: IPortfolio, product: IProduct): string {
+    const associationKey = hashValues(portfolio.node.addr, product.node.addr);
```

Comment:
I'm wondering - shouldn't the hash also include the Stack of the product? What if I associate a Portfolio with 2 Products, one from the same Stack the Portfolio is in, and one from a different Stack? Won't I get the same IDs if it happens that the 2 Products have the same addresses?

Reply:
I might've erroneously removed this when removed the scope argument but added it back.   Kind of awkward to test unfortunately.